File: app/src/androidTest/java/com/paradigmadigital/dagger/ui/main/MainActivityTest.java
package com.paradigmadigital.dagger.ui.main;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import static junit.framework.Assert.assertEquals;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {

    @Rule
    public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void useAppContext() throws Exception {

        assertEquals("Test collaborator version", activityRule.getActivity().getTitle());
    }
}

File: app/src/androidTest/java/eu/sesma/dagger/ui/main/MainActivityTest.kt
package eu.sesma.dagger.ui.main

import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4

import junit.framework.Assert.assertEquals

@RunWith(AndroidJUnit4::class)
class MainActivityTest {

    @Rule
    var activityRule = ActivityTestRule(MainActivity::class.java)

    @Test
    @Throws(Exception::class)
    fun useAppContext() {

        assertEquals("Test collaborator version", activityRule.activity.title)
    }
}
